### ISSUE TITLE
CTRL + Enter should submit

### ISSUE THREAD

**Issue by Zer0t3ch** [voat/voat#455]
When typing a comment (or maybe a post) CTRL+Enter should be the equivalent of clicking the submit button. (Though I suppose implementing this on posts would be useless because of the captcha)

I believe the implementation would be nothing more than a bit of HTML or JS, but I'm not certain about that.

**Comment by pharno** [comment]
+1

**Comment by yotamN2** [comment]
+1

Shouldn't be too hard to implement, I think the site should be completely keyboard accessible

**Comment by yesiamquitesmall** [comment]
I assume this would come under the already existing request #54

**Comment by Zer0t3ch** [comment]
@yesiamquitesmall It could, but they probably shouldn't be merged. This isn't really about navigation, and it should be just a quick 5-minute tweak. #54 would probably require a decent bit a forethought as well as a good bit of time implementing. (As it's more than just a quick HTML modification)

**Comment by yotamN2** [comment]
Should it post only when the input have focus or it doesn't matter?

**Comment by jksg** [comment]
@yotamN2 Hm? It *must* submit only the form with focus, since you can have multiples of the reply form visible (1 for each existing comment on the page).

Anyway, anyone who uses keyboard extensively / exclusively can already tab to move between controls.

**Comment by Zer0t3ch** [comment]
@jksg You mean using tab to select the submit button?

**Comment by yotamN2** [comment]
I created a script that post on ctrl+enter keys but I don't know how to implement it in the code, I want to create a new file for this because there is no reason to load in on the front page for example.

**Comment by jksg** [comment]
@Zer0t3ch Yes. `Tab` + `space` to “click” the button. That is how I submit all my text forms on all sites, including this one :)

A custom site-specific solution like adding a behaviour to Ctrl-Enter is perhaps *convenient*, but not more *accessible*.

**Comment by yotamN2** [comment]
@jksg `tab` + `space` sound confusing, I mean there can always be both.

**Comment by jksg** [comment]
@yotamN2  Yes, I’m not arguing against this feature. I’m explaining that there already exists full keyboard controls in every browser for accessibility purposes.

Tab is not confusing at all. It cycles through every focusable element on a page. And it is universal (unless a website deliberately breaks it with JS). Having to remember which website has a custom shortcut for submitting forms, on the other hand, is quite confusing ;)

**Comment by Zer0t3ch** [comment]
@jksg I got the idea from Reddit, and you would be amazed how much more convenient it can make the process of commenting. Anyway, it's not a special thing, IIRC, it's something like this:

If you have a single-line text form, you can use HTML to specify which button get's "clicked" upon the enter key being pressed. Ctrl+Enter is the same, except for multiline text forms. I believe it's a really simple and standardized thing.

If I'm wrong, then maybe people going out of their way to make it a reality is a waste of time, but if it really is just a minor HTML tweak like I think it is, then there really doesn't seem to be any reason *not* to add it. People who don't want to use it don't have to.

**Comment by jksg** [comment]
@Zer0t3ch You cannot specify which button gets activated with Enter using HTML. Browsers typically activate the first button when you hit Enter on any input element  within the form (submit button, text field, radio buttons, checkboxes etc.), but there is surprisingly no standard spec on this! Which is why as a heavy keyboard user I much prefer to tab to the specific button I want to activate instead of hoping the form submits (I’ve seen forms activate the reset/cancel button on Enter…). The exception is a textarea (multiline text box) where Enter inserts a return.

You also cannot specify the behaviour of Ctrl-Enter (or any other key combination) in HTML, although it is quite trivial to implement in JavaScript — more or less a single line of code in jQuery which Voat uses. I agree it’ll be a nice convenience feature for those that want it.

**Comment by Zer0t3ch** [comment]
@jksg Ok, good to get some information from someone who knows what they're talking about! xD I haven't actively done any web dev in a couple years, and that was never something that I really dove into.

**Comment by x0x7** [comment]
@yotamN do you still have that script.  I may try posting it into [Amateur-Voat-Enhancements](https://github.com/HorzaGobuchul/).  It's not even my project so if you have code that will make that easier I might do it.
